runtime/lib/backend/null_qubit/NullQubit.hpp

```diff
@@ -40,14 +69,42 @@ namespace Catalyst::Runtime::Devices {
  *   of the device; these are used to implement Quantum Instruction Set (QIS) instructions.
  */
 struct NullQubit final : public Catalyst::Runtime::QuantumDevice {
-    NullQubit(const std::string &kwargs = "{}") {}
-    ~NullQubit() = default; // LCOV_EXCL_LINE
+    NullQubit(const std::string &kwargs = "{}")
+    {
+        auto device_kwargs = Catalyst::Runtime::parse_kwargs(kwargs);
+        if (device_kwargs.find("track_resources") != device_kwargs.end()) {
+            track_resources_ = device_kwargs["track_resources"] == "True";
+        }
+    }
+    ~NullQubit()
+    {
+        if (this->track_resources_) {
+            PrintResourceUsage();
```

Comment:
Another thing I found is that having the printing in the destructor means that we will accumulate counts for all qnode executions within a session. That could be what the user is after, but for me I was looking for the resources of a single qnode execution so added the printing and resource reset in the release all qubits function.

Reply:
This idea sounds good to me, any input @maliasadi ?

Reply:
Fixed in [41f8d28](https://github.com/PennyLaneAI/catalyst/pull/1619/commits/41f8d2898e0983d7e7a6c7353236a99a6b7d8a34)